src/lib/es2018.asyncgenerator.d.ts

```diff
@@ -2,9 +2,9 @@
 
 interface AsyncGenerator<T = unknown, TReturn = any, TNext = unknown> extends AsyncIterator<T, TReturn, TNext> {
     // NOTE: 'next' is defined using a tuple to ensure we report the correct assignability errors in all places.
-    next(...args: [] | [TNext]): Promise<IteratorResult<T, TReturn>>;
-    return(value: TReturn | PromiseLike<TReturn>): Promise<IteratorResult<T, TReturn>>;
-    throw(e: any): Promise<IteratorResult<T, TReturn>>;
+    next(...args: [] | [TNext]): Promise<IteratorResult<T, TReturn extends undefined ? TReturn : TReturn extends PromiseLike<infer UReturn> ? UReturn : TReturn>>;

```

Comment:
I am wary about using conditional types here, as this could cause issues with assignability to `AsyncGenerator`.

Reply:
:+1: Removed.